### PR TITLE
Update roctx.cpp

### PR DESCRIPTION
**Pull request by rkebichi**
Added roctxRangePushA and roctxRangePop fcts

### CODE REVIEW

File: src/roctx/roctx.cpp [hunk 1]
```diff
@@ -114,13 +117,30 @@ PUBLIC_API void roctxMarkA(const char* message) {
...
+  roctx::cb_table.get(ROCTX_API_ID_roctxRangePushA, &api_callback_fun, &api_callback_arg);
+  if (api_callback_fun) api_callback_fun(ACTIVITY_DOMAIN_ROCTX, ROCTX_API_ID_roctxRangePushA, &api_data, api_callback_arg);
+  message_stack.push(strdup(message));
   API_METHOD_SUFFIX
 }
 
 PUBLIC_API int roctxRangePop() {
   API_METHOD_PREFIX
-  EXC_ABORT(ROCTX_STATUS_ERROR, "method is not implemented");
+  //EXC_ABORT(ROCTX_STATUS_ERROR, "method is not implemented");
```
Comment (eshcherb) [inline src/roctx/roctx.cpp]: Please remove dead code

File: src/roctx/roctx.cpp [hunk 2]
```diff
@@ -114,13 +117,30 @@ PUBLIC_API void roctxMarkA(const char* message) {
 
 PUBLIC_API int roctxRangePushA(const char* message) {
   API_METHOD_PREFIX
-  EXC_ABORT(ROCTX_STATUS_ERROR, "method is not implemented");
+  //EXC_ABORT(ROCTX_STATUS_ERROR, "method is not implemented");
```
Comment (eshcherb) [inline src/roctx/roctx.cpp]: Please remove dead code

File: src/roctx/roctx.cpp [hunk 3]
```diff
@@ -114,13 +117,30 @@ PUBLIC_API void roctxMarkA(const char* message) {
...
+  roctx_api_data_t api_data{};
+  api_data.args.roctxRangePushA.message = strdup(message);
+  activity_rtapi_callback_t api_callback_fun = NULL;
+  void* api_callback_arg = NULL;
+  roctx::cb_table.get(ROCTX_API_ID_roctxRangePushA, &api_callback_fun, &api_callback_arg);
+  if (api_callback_fun) api_callback_fun(ACTIVITY_DOMAIN_ROCTX, ROCTX_API_ID_roctxRangePushA, &api_data, api_callback_arg);
+  message_stack.push(strdup(message));
   API_METHOD_SUFFIX
```
Comment (eshcherb) [inline src/roctx/roctx.cpp]: The function should return '-1' on error.
Could you please change API_METHOD_SUFFIX on API_METHOD_CATCH(-1)

File: src/roctx/roctx.cpp [hunk 4]
```diff
@@ -114,13 +117,30 @@ PUBLIC_API void roctxMarkA(const char* message) {
...
   API_METHOD_PREFIX
-  EXC_ABORT(ROCTX_STATUS_ERROR, "method is not implemented");
+  //EXC_ABORT(ROCTX_STATUS_ERROR, "method is not implemented");
+  roctx_api_data_t api_data{};
+  api_data.args.roctxRangePushA.message = strdup(message);
+  activity_rtapi_callback_t api_callback_fun = NULL;
+  void* api_callback_arg = NULL;
+  roctx::cb_table.get(ROCTX_API_ID_roctxRangePushA, &api_callback_fun, &api_callback_arg);
+  if (api_callback_fun) api_callback_fun(ACTIVITY_DOMAIN_ROCTX, ROCTX_API_ID_roctxRangePushA, &api_data, api_callback_arg);
+  message_stack.push(strdup(message));
```
Comment (eshcherb) [inline src/roctx/roctx.cpp]: Please add 'return 0;' to return '0' on success.

File: src/roctx/roctx.cpp [hunk 5]
```diff
@@ -114,13 +117,30 @@ PUBLIC_API void roctxMarkA(const char* message) {
...
+  roctx_api_data_t api_data{};
+  activity_rtapi_callback_t api_callback_fun = NULL;
+  void* api_callback_arg = NULL;
+  roctx::cb_table.get(ROCTX_API_ID_roctxRangePop, &api_callback_fun, &api_callback_arg);
+  if (api_callback_fun) api_callback_fun(ACTIVITY_DOMAIN_ROCTX, ROCTX_API_ID_roctxRangePop, &api_data, api_callback_arg);
+  if (message_stack.empty()) {
+      EXC_ABORT(ROCTX_STATUS_ERROR, "Pop from empty stack!");
+  } else {
+      message_stack.pop();
+  }
```
Comment (eshcherb) [inline src/roctx/roctx.cpp]: Please add 'return 0;' to return '0' on success

File: src/roctx/roctx.cpp [hunk 6]
```diff
@@ -114,13 +117,30 @@ PUBLIC_API void roctxMarkA(const char* message) {
...
   API_METHOD_SUFFIX
```
Comment (eshcherb) [inline src/roctx/roctx.cpp]: The function should return '-1' on error.
Could you please change API_METHOD_SUFFIX on API_METHOD_CATCH(-1)

File: test/MatrixTranspose_test/MatrixTranspose.cpp [hunk 7]
```diff
@@ -97,14 +100,23 @@ int main() {
...
     
+        roctxMarkA("before hipLaunchKernel");
+        roctxRangePushA("hipLaunchKernel");
+        //roctxRangePushA("before hipLaunchKernel");
+        // roctxRangePushA("before hipLaunchKernel");
         // Lauching kernel from host
         hipLaunchKernel(matrixTranspose, dim3(WIDTH / THREADS_PER_BLOCK_X, WIDTH / THREADS_PER_BLOCK_Y),
                         dim3(THREADS_PER_BLOCK_X, THREADS_PER_BLOCK_Y), 0, 0, gpuTransposeMatrix,
                         gpuMatrix, WIDTH);
+        //roctxRangePop();
```
Comment (eshcherb) [inline test/MatrixTranspose_test/MatrixTranspose.cpp]: Dead code

File: test/MatrixTranspose_test/MatrixTranspose.cpp [hunk 8]
```diff
@@ -97,14 +100,23 @@ int main() {
         // Memory transfer from host to device
         hipMemcpy(gpuMatrix, Matrix, NUM * sizeof(float), hipMemcpyHostToDevice);
     
+        roctxMarkA("before hipLaunchKernel");
+        roctxRangePushA("hipLaunchKernel");
+        //roctxRangePushA("before hipLaunchKernel");
+        // roctxRangePushA("before hipLaunchKernel");
```
Comment (eshcherb) [inline test/MatrixTranspose_test/MatrixTranspose.cpp]: Dead code

File: src/roctx/roctx.cpp [hunk 9]
```diff
@@ -61,6 +62,8 @@ THE SOFTWARE.
   (void)err;                                                                                       \
   return X;
 
+std::stack<std::string> message_stack;
```
Comment (eshcherb) [inline src/roctx/roctx.cpp]: Please make it thread local as the ranges should be per CPU threads:
static thread_local std::stack<std::string> message_stack;

File: src/roctx/roctx.cpp (hunk 4 above)
Comment (rkebichi) [reply src/roctx/roctx.cpp]: done

File: src/roctx/roctx.cpp (hunk 5 above)
Comment (rkebichi) [reply src/roctx/roctx.cpp]: done

**Pull request merged**